_docs/developer/testing/cypress.md

```diff
@@ -0,0 +1,187 @@
+GET /home 200 287.217 ms - -
+GET /authentication/login 200 2.297 ms - -
+POST /authentication/check_login?old=http%253A%252F%252Flocalhost%253A1501%252Fhome 302 2772.070 ms - 96
+GET /home 302 210.507 ms - 210
+GET /authentication/login?old=http%3A%2F%2Flocalhost%3A1501%2Fhome 200 264.544 ms - -
+GET /authentication/login 200 1.629 ms - -
+POST /authentication/check_login?old=http%253A%252F%252Flocalhost%253A1501%252Fhome 302 205.710 ms - 96
+```
+
+A complete overview can be found [here](https://docs.cypress.io/guides/core-concepts/test-runner.html#Overview)
+
+### Running Headless
+
+You can run the entire test suite in headless mode. This will pipe the browser display directly to Cypress without getting displayed and is faster than the test-runner. This can be done with the command:
+
+```bash
+# from the site directory
+npx cypress run 
+```
+
+If any test fails a screen shot will be taken of the step that failed. Additionally, a video will be recorded each time of the run. They can be found under the `site/cypress/screenshots` and `site/cypress/videos` directories.
+
+See the full usage [here](https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases)
+
+## Writing Tests
+
+Test files are written under the `site/cypress/integration` and should follow the format `*.spec.js*` you can
+create folders as needed to organize groups of tests together.
+
+**Cypress test suite structure:**
+
+```
+cypress.json
+cypress/
+├── fixtures
+│   └── example.json
+├── integration
+│   └── ...
+│
+├── plugins
+│   └── index.js
+├── screenshots
+│ 
+├── support
+│   ├── commands.js
+│   └── index.js
+└── videos
+   
+```
+
+While the main test files are in the integration directory, there are some supporting files. The `fixtures` directory contains files used to mock data. This can be text files, JSON, images, etc. 
+
+The `commands.js` file 
+allow you to program Cypress commands. Actions that require multiple steps and might be done often can be grouped together under a single command to prevent repetition, for example the action of logging into the Submitty site is a command. You can also write plain JavaScript functions in this file and import them using ES6 module syntax.
+
+The `support/index.js` file instructs Cypress which files to include with every test, you most likely won't need to touch this file. 
+
+The `plugins/index.js` allows you to extend Cypress's default behavior. You can read more about them [here](https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases). 
+
+The top level `cypress.json` is the global configuration used.
+
+### Test Structure
+
+Cypress borrow's Mocha's Behavior-driven development (BDD) [test interface](https://mochajs.org/#bdd) to organize tests. 
+
+You will typically use `describes`, `context`, and `it` to organize tests. `describes` and `context` organize tests into groups, either or can be used they both act the same. Here is an example:
```

Comment:
```suggestion
You will typically use `describes` and `it` to organize tests. `describes` organize tests into groups, either or can be used they both act the same.
To test assertions, you will use the integrated [Chai tool](https://docs.cypress.io/guides/references/bundled-tools.html#Chai) to write your assertions. Here, you will use the `expect()` function wrapped around an object, and use a series of [assertions](https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions) about it.

Here is an example:
```

Reply:
You mean `describes` and `context` right, `it` definitely has a different behavior with cypress, you can't write any commands or assertions inside of a `describes` or `context` block, only within an `it`